tests/example_nft_tests.cdc

```diff
@@ -0,0 +1,227 @@
+
+    Test.assertEqual(expectedCollectionLength, actualCollectionIDs.length)
+}
+
+access(all) fun testTransferNFT() {
+
+    let nftIDs = scriptExecutor("get_collection_ids.cdc", [
+            recipient.address,
+            /public/cadenceExampleNFTCollection
+        ]) as! [UInt64]? ?? panic("Could not get collection IDs from admin")
+    let expectedTransferID = nftIDs[0]
+
+    txExecutor("transfer_nft.cdc", [recipient], [admin.address, "ExampleNFT", admin.address, expectedTransferID], nil, nil)
+
+    var typ = CompositeType(buildTypeIdentifier(admin, "NonFungibleToken", "Transfer"))!
+    Test.assertEqual(1, blockchain.eventsOfType(typ).length)
+
+    let adminIDs = scriptExecutor("get_collection_ids.cdc", [
+            admin.address,
+            /public/cadenceExampleNFTCollection
+        ]) as! [UInt64]? ?? panic("Could not get collection IDs from admin")
+    let actualTransferID = adminIDs[0]
+
+    Test.assertEqual(expectedTransferID, actualTransferID)
+}
+
+access(all) fun testTransferMissingNFT() {
+    let expectedErrorMessage = "The collection does not contain the specified ID"
+    let expectedErrorType = ErrorType.TX_PRE
+
+    txExecutor(
+        "transfer_nft.cdc",
+        [recipient],
+        [admin.address, "ExampleNFT", admin.address, 10 as UInt64],
+        expectedErrorMessage,
+        expectedErrorType
+    )
+}
+
+access(all) fun testBorrowNFT() {
+    txExecutor(
+        "mint_nft.cdc",
+        [admin], [
+            recipient.address,
+            "NFT Name",
+            "NFT Description",
+            "NFT Thumbnail",
+            [0.05],
+            ["Creator Royalty"],
+            [admin.address]
+        ], nil,
+        nil
+    )
+    let nftIDs = scriptExecutor("get_collection_ids.cdc", [
+            recipient.address,
+            /public/cadenceExampleNFTCollection
+        ]) as! [UInt64]? ?? panic("Could not get collection IDs")
+    let mintedID = nftIDs[0]
+
+    // Panics if not successful - enough to run the script here
+    let scriptResult = scriptExecutor("borrow_nft.cdc", [recipient.address, mintedID])
+}
+
+access(all) fun testBorrowMissingNFT() {
+    expectScriptFailure("borrow_nft.cdc", [admin.address, 10 as UInt64])
+}
+
+access(all) fun testGetCollectionIDs() {
+    let expectedCollectionLength = 1
+    
+    let actualNFTIDs = scriptExecutor("get_collection_ids.cdc", [
+            recipient.address,
+            /public/cadenceExampleNFTCollection
+        ]) as! [UInt64]? ?? panic("Could not get collection IDs")
+
+    Test.assertEqual(expectedCollectionLength, actualNFTIDs.length)
+}
+
+access(all) fun testGetCollectionLength() {
+    let expectedCollectionLength = 1
+
+    let actualCollectionLength = scriptExecutor("get_collection_length.cdc", [admin.address]) as! Int?
+        ?? panic("Could not get collection length")
+
+    Test.assertEqual(expectedCollectionLength, actualCollectionLength)
+}
```

Comment:
Are these basically the same test?

Reply:
I know we're using different scripts, but we should probably be making sure the IDs are correct in the first one instead of just checking the length

Reply:
These could probably be consolidated into one test, but they do use different scripts since one now uses .getIDs() and the other uses .getLength().

Not sure how we would validate the id given it's the uuid. Maybe confirming NFT.uuid == NFT.getID() && collection.getIDs().contains(id)?